gson/src/main/java/com/google/gson/JsonDeserializer.java

```diff
@@ -51,7 +51,9 @@
  * class IdDeserializer implements JsonDeserializer&lt;Id&gt;() {
  *   public Id deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
  *       throws JsonParseException {
- *     return new Id((Class)typeOfT, id.getValue());
+ *     JsonObject jsonObject = json.getAsJsonObject();
+ *     long idValue = jsonObject.get("value").getAsLong();
```

Comment:
Maybe this should use `getAsJsonPrimitive​` to make the logic here a bit stricter, otherwise it would for example also accept a JSON array (see [`JsonArray.getAsLong()`](https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/JsonArray.html#getAsLong())) which is probably not desired.
```suggestion
 *     long idValue = jsonObject.getAsJsonPrimitive​("value").getAsLong();
```

Reply:
Thanks for the suggestion. updated

Reply:
Thanks, but maybe it would be better to replace it with what I mentioned in https://github.com/google/gson/pull/2192#discussion_r967005649 (sorry for the confusion).